### PR TITLE
Handle package indexes update fails

### LINKED ISSUE #1506: Update indexes in the IDE silently fails
### Steps to reproduce
1. Add broken URL into `$WS/__packages__/extra.txt`
2. Open IDE
3. Open deploy popup and click "Update indexes"

### Expected behavior
The snackbar error message should be shown.

### Actual behavior
Nothing.

### PR DESCRIPTION
**Pull request by brusherru**
Now it will show an error message and remove new URLs from `.cli-config.yml` that could cause errors (but left downloaded files in the `__packages__` directory to make possible to inspect what is really went wrong).
So IDE will be usable after the error.

It fixes #1506

### CODE REVIEW

**Comment by brusherru** [comment]
@nkrkv updated. Check it out, please!

**Comment by brusherru** [comment]
@nkrkv check it out, please!

File: packages/arduino-cli/src/config.js [hunk 1]
```diff
@@ -50,10 +55,31 @@ export const addPackageIndexUrls = (configPath, urls) =>
...
+// :: Path -> [URL] -> Promise [URL] Error
+export const removePackageIndexUrls = (configPath, urls) => {
+  let removedUrls = [];
+
+  return fse
+    .readFile(configPath, { encoding: 'utf8' })
+    .then(YAML.parse)
+    .then(cfg => {
+      removedUrls = R.compose(
+        R.filter(R.contains(R.__, urls)),
```
Comment (evgenykochetkov) [inline packages/arduino-cli/src/config.js]: `R.intersection(urls)`?

File: packages/arduino-cli/src/config.js [hunk 2]
```diff
@@ -50,10 +55,31 @@ export const addPackageIndexUrls = (configPath, urls) =>
...
+        R.path(['board_manager', 'additional_urls'])
+      )(cfg);
+      return overAdditionalUrls(
+        R.pipe(R.defaultTo([]), R.reject(R.contains(R.__, urls)))
```
Comment (evgenykochetkov) [inline packages/arduino-cli/src/config.js]: `R.difference(R.__, removedUrls)`?

**Comment by brusherru** [comment]
@evgenykochetkov fixed! Check it out, please

**Comment by brusherru** [comment]
@evgenykochetkov fixed! Check it out, please

**Comment by brusherru** [comment]
@evgenykochetkov @nkrkv check it out, please!
I think now it's much better :)

File: packages/arduino-cli/src/listAvailableBoards.js [hunk 3]
```diff
@@ -14,24 +14,35 @@
...
 /**
- * Finds all package index json files in the specified directory.
- * Returns a promise with a list of full paths to the json files.
+ * Returns a list of paths to the additional package index files.
+ *
+ * Gets filenames of additional package index files from arduino cli config
+ * by parsing URLs and joins filenames with path to packages directory.
  *
- * :: Path -> Promise [Path] Error
+ * :: (_ -> Object) -> Path -> Promise [Path] Error
```
Comment (nkrkv) [inline packages/arduino-cli/src/listAvailableBoards.js]: The signature is

```
:: (() -> Promise Object Error) -> Path -> Promise [Path] Error
```

File: packages/arduino-cli/src/listAvailableBoards.js [hunk 4]
```diff
@@ -77,10 +88,10 @@ const getAvailableBoards = R.compose(
  * Reads all package index json files in the specified directory
  * and returns a promise with a list of Available Boards.
  *
- * :: Path -> Promise [AvailableBoard] Error
+ * :: (_ -> Object) -> Path -> Promise [AvailableBoard] Error
```
Comment (nkrkv) [inline packages/arduino-cli/src/listAvailableBoards.js]: ditto

**Comment by brusherru** [comment]
@nkrkv fixed!

**Pull request merged**